### ISSUE TITLE
compatible to termux_terminal?

### ISSUE THREAD

**Issue by ApolloGods** [Genymobile/scrcpy#3696]
hello guys,

i`ve installed Win 11 & Android 13 /API 33 - Win11 every fine (wrote a script.bat to bind and open scrcpy) **.bat
Android App scrcpy also fine ;-)
but in Termux (installed termux sdl & x11-repo) it starting process but ends up in

~ $ scrcpy -s 192.168.1.100:5555
scrcpy 1.25 https://github.com/Genymobile/scrcpy
DEBUG: ADB device found:
DEBUG: --> (tcpip) 192.168.1.100:5555 device SM_G986B
DEBUG: (tcpip) emulator-5554 device SM_G986B
DEBUG: Device serial: 192.168.1.100:5555
DEBUG: Using server: /data/data/com.termux/files/usr/share/scrcpy/scrcpy-server
/data/data/com.termux/files/usr/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 49.3 MB/s (42151 bytes in 0.001s)
[server] INFO: Device: samsung SM-G986B (Android 13)
DEBUG: Server connected
DEBUG: Starting controller thread
DEBUG: Starting receiver thread
[server] DEBUG: Using encoder: 'OMX.Exynos.AVC.Encoder'
INFO: Renderer: opengl
INFO: OpenGL version: 4.5 (Compatibility Profile) Mesa 22.3.3
INFO: Trilinear filtering enabled
DEBUG: Using icon: /data/data/com.termux/files/usr/share/icons/hicolor/256x256/apps/scrcpy.png
INFO: Initial texture: 1080x2400
DEBUG: Starting demuxer thread

and no screen - had i do to add options like  | ffplay (have installed it to termux to can do it but don´t know how) ? ***

in App Bugjaeger scrcpy goes to Camview (very useful) - also record is possible

i didn´t get it running by myself in manual mode - maybe to do it with tasker

thx for help

**.bat
@echo Off
cd C:/PATH/to/platform-tools/(Android Studio or platform-tools only for itself / Android -> https://github.com/rendiix/termux-adb-fastboot
echo startup adb-server
adb kill-server
adb start-server
adb devices
adb connect HOST:PORT (192.168.1.1:5555)
timeout 5
scrcpy -s HOST:PORT

***
copy ffplay to /data/data/com.termux/files/usr/bin
cp example/SDL2/ffplay/ffplay /data/data/com.termux/files/usr/bin
chmod 755 /data/data/com.termux/files/usr/bin/ffplay

play video
ffplay -i /sdcard/video/test.mp4

play music whit 1.5x speed
ffplay -af atempo=1.5 -i /sdcard/music/hello.flac

2x speed playback
ffplay -af atempo=2.0 -vf setpts=1/2*PTS -i /sdcard/video/test.mp4

 play at any speed: atempo=x setpts=1/x*PTS

no window title(set title is null)
ffplay -window_title -i /sdcard/video/test.mp4

full screen
ffplay -fs -i /sdcard/video/test.mp4

**Comment by eeriemyxi** [comment]
> also record is possible
You mean to say that when you try to record your screen using SCRCPY on Termux, it works?

**Comment by eeriemyxi** [comment]
> yes in App Bugjaeger


Well I already have WSLg and WSL working, so I have motivation to control my own phone using my own phone, but I think the issue is with your Xserver or VNC server, or VNC server viewer app that you have installed on your phone. Read and understand the steps mentioned above and if you manage to get it working—well... I don't even know.

**Comment by s3sahni** [comment]
bro just use termux-x11 it will support scrcpy "it even works on my 2gb ram calculator" i used to record my gaming videos with it.

**Comment by datsoy** [comment]
How to make scrcpy use termux-x11? Care to tell for this uncle? Is that possible?

I managed to deploy a redroid in my oracle arm instance and able to connect via adb, scrcpy 2.1.1 is available in termux, now I would like to see/control it using my phone via termux-x11. How to make it work? Thank you.

**Comment by s3sahni** [comment]
yeah its possible you can use it, actually i use it in my daily life to record my gameplay videos often ly

its perfectly supported ^_^ !

just use to install & use xterm first

you will also need to install android-tools by using apt , ti would work fine with wireless debugging 

if you wish to use scrcpy with otg cable , you will need termux-adb from github 

firstly, make sure to connect usb/wireless debugging from host phone to target phone

then setup termux-x11, like this

termux-x11 -xstartup "xterm" 
(make sure you have xterm installed)

then in termux x11 app you will have xterm running,
just execute scrcpy and done

( note: scrcpy will only work when adb is connected to a device )

**Comment by ngothanhnam187** [comment]
I can start scrcpy in x11. But scrcpy is not full screen of x11. How to configuration it?
Many thanks 
![Screenshot_2023-12-18-20-26-56-745_com termux x11](https://github.com/Genymobile/scrcpy/assets/154074724/6c220ea1-8512-4722-993f-ed693c61c171)

**Comment by RebornOfc** [comment]
> I can start scrcpy in x11. But scrcpy is not full screen of x11. How to configuration it? Many thanks ![Screenshot_2023-12-18-20-26-56-745_com termux x11](https://private-user-images.githubusercontent.com/154074724/291298254-6c220ea1-8512-4722-993f-ed693c61c171.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNTAxNDMsIm5iZiI6MTcwNDI0OTg0MywicGF0aCI6Ii8xNTQwNzQ3MjQvMjkxMjk4MjU0LTZjMjIwZWExLTg1MTItNDcyMi05OTNmLWVkNjkzYzYxYzE3MS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwM1QwMjQ0MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Yjg0OWFhYTUxOTU2YmVmNTI0YTlhZDgyMTA0OTAzMmEzM2Y2MDVmOTYyNGU1NTI1NDgzOGE5MTFjOGJiYzMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.feG-TKYlbRPmtWKi9zKY2dOcZ-3i0Yk1Jx2DT3jM7YU)

Hey, I have a question. How in the world you are in termux-x11 but the scrcpy window is casting a game? Was not it supposed to be casting an infinite loop of termux-x11?

**Comment by ngothanhnam187** [comment]
> > I can start scrcpy in x11. But scrcpy is not full screen of x11. How to configuration it? Many thanks ![Screenshot_2023-12-18-20-26-56-745_com termux x11](https://private-user-images.githubusercontent.com/154074724/291298254-6c220ea1-8512-4722-993f-ed693c61c171.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNTAxNDMsIm5iZiI6MTcwNDI0OTg0MywicGF0aCI6Ii8xNTQwNzQ3MjQvMjkxMjk4MjU0LTZjMjIwZWExLTg1MTItNDcyMi05OTNmLWVkNjkzYzYxYzE3MS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwM1QwMjQ0MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Yjg0OWFhYTUxOTU2YmVmNTI0YTlhZDgyMTA0OTAzMmEzM2Y2MDVmOTYyNGU1NTI1NDgzOGE5MTFjOGJiYzMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.feG-TKYlbRPmtWKi9zKY2dOcZ-3i0Yk1Jx2DT3jM7YU)
> 
> Hey, I have a question. How in the world you are in termux-x11 but the scrcpy window is casting a game? Was not it supposed to be casting an infinite loop of termux-x11?
> 
> (Now that I think of, maybe that is another device?)

Yeah. I running game/ app in another device. scrcpy is best with low lantency compare with teamviewer, anydesk, ...
It not native as SCRCPY in window but it is still good enough.

**Comment by ngothanhnam187** [comment]
> > I can start scrcpy in x11. But scrcpy is not full screen of x11. How to configuration it? Many thanks ![Screenshot_2023-12-18-20-26-56-745_com termux x11](https://private-user-images.githubusercontent.com/154074724/291298254-6c220ea1-8512-4722-993f-ed693c61c171.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5MDg4MDQsIm5iZiI6MTcwMjkwODUwNCwicGF0aCI6Ii8xNTQwNzQ3MjQvMjkxMjk4MjU0LTZjMjIwZWExLTg1MTItNDcyMi05OTNmLWVkNjkzYzYxYzE3MS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxOFQxNDA4MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWU4NmZlNjg3MjVlNzczYzE2OTAzNjYyZTliMDE3NWYxMWRjMzdhNDM4MDM0NGQ0ZmZlYTZjZjYxYTVhNjEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.R6L1XKhTdexOdwdXtVvF_7weJp7NaITgU9A39pghZ44)
> 
> bro its so simple you must have installed any window manager like fluxbox or jwm with apt in termux so you can easily switch to fullscreen mode by clicking on maximize button;
> 
> you can use xterm within fluxbox for more convenience :)

Thx you so much. I will try it.

**Comment by mylove90** [comment]
This would be beautiful if it is used with an Android tv box that doesn't have chromecast option built-in with an old Android version that gets treated as a tablet 

Please give tutorial if possible

**Comment by konsloiz** [comment]
Do you know how can I install termux-x11 on termux running on an Android TV box?

**Comment by Eated-Universe** [comment]
Nice

**Comment by DewandaArya** [comment]
so it work on termuxx11 but not tigervnc vncserver with rvnc viewer?
